cli/cmd/install.go

```diff
@@ -553,6 +554,19 @@ func (options *installOptions) allStageFlagSet() *pflag.FlagSet {
 		"Restrict the Linkerd Dashboard's default privileges to disallow Tap and Check",
 	)
 
+	return flags
+}
+
+// allStageFlagSetButNotRecordable returns flags usable for single and multi-stage  installs and
+// upgrades but which are not recordable. For multi-stage installs, users must set these flags consistently
+// across commands.
+func (options *installOptions) allStageFlagSetButNotRecordable() *pflag.FlagSet {
+	flags := pflag.NewFlagSet("all-stage", pflag.ExitOnError)
```

Comment:
This is the same label as in `allStageFlagSet()`. Not sure if that clashes, but better to give it a different one.